lib/compiler.ts

```diff
@@ -228,3 +233,71 @@ main() async {
     return text;
   }
 }
+
+export class BrowserCompiler extends Compiler {
+  private server?: http.Server;
+  private browser?: playwright.Browser;
+  private page?: playwright.Page;
+
+  public static async start(serverRoot: string) {
+    const compiler = new BrowserCompiler();
+    compiler.server = http.createServer((request, response) => {
+      return handler(request, response, {
+        public: serverRoot,
+      });
+    });
+    compiler.server.listen();
+
+    compiler.browser = await playwright.chromium.launch();
+    const {port} = compiler.server.address() as AddressInfo;
+    compiler.page = await compiler.browser.newPage();
+    await compiler.page.goto(`http://localhost:${port}`);
```

Comment:
@jgerigmeyer @nex3 what do you think of this approach (and the instructions in README)? Is there a better alternative to requiring an HTML file and spinning up a server?